### ISSUE TITLE
branch-3.0 crash on sort aggregate

### ISSUE THREAD

**Issue by chaoyli** [StarRocks/starrocks#20986]
```
CREATE TABLE ACTIVITY2 AS SELECT -10 AS case_id, 1 AS activity UNION ALL SELECT -10 AS case_id, 2 AS activity;

set enable_sort_aggregate = true;
set enable_tablet_internal_parallel = false;
SELECT ARRAY_AGG(`activity`) AS `agg_activity`,
       ANY_VALUE(`case_id`) AS `case_id` FROM `ACTIVITY2` GROUP BY `case_id`;
```

**Comment by chaoyli** [comment]
```
*** Aborted at 1680552805 (unix time) try "date -d @1680552805" if you are using GNU date ***
PC: @          0x451633b starrocks::AggregateFunctionBatchHelper<>::merge_batch()
*** SIGSEGV (@0x0) received by PID 1582 (TID 0x7fe57a7bb640) from PID 0; stack trace: ***
    @          0x6ffd1aa google::(anonymous namespace)::FailureSignalHandler()
    @     0x7fe6057d5520 (unknown)
    @          0x451633b starrocks::AggregateFunctionBatchHelper<>::merge_batch()
    @          0x3bcb47f starrocks::SortedStreamingAggregator::_update_states()
    @          0x3bcc7a0 starrocks::SortedStreamingAggregator::streaming_compute_agg_state()
    @          0x3bd9407 starrocks::pipeline::SortedAggregateStreamingSinkOperator::push_chunk()
    @          0x39fe7d4 starrocks::pipeline::PipelineDriver::process()
    @          0x5bdefd0 starrocks::pipeline::GlobalDriverExecutor::_worker_thread()
    @          0x5f308db starrocks::ThreadPool::dispatch_thread()
    @          0x5f2ab9a starrocks::Thread::supervise_thread()
    @     0x7fe605827b43 (unknown)
    @     0x7fe6058b8bb4 clone
    @                0x0 (unknown)
```

**Comment by chaoyli** [comment]
The commit hash is 55f25466

**Comment by stdpain** [comment]
resolved in aa3f5b680816e24cc77262374e7e1128bc0bf86c
